src/lib/transaction_snark/transaction_snark.ml

```diff
@@ -1844,21 +1958,112 @@ module Base = struct
                party or party as a stack implies a hash collision.
             *)
             let is_some = Field.equal h h' in
-            let data = ((party_or_stack, hd_r), (stack, tl_r)) in
+            let data : t * t =
+              ( { hash = party_or_stack; data = hd_r }
+              , { hash = stack; data = tl_r } )
+            in
             { Snapp_basic.Flagged_option.is_some; data }
+        end
 
-          let push_stack ((h_hd, r_hd) : t) ~onto:((h_tl, r_tl) : t) : t =
-            let h = hash_cons h_hd h_tl in
+        module Call_stack
+            (*: Parties_logic.Call_stack_intf
```

Comment:
Might it be worth uncommenting this constraint and adding `with type t = (Party.t * unit, Parties.Digest.t) Parties.Party_or_stack.t list` to make it externally usable?